src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -1207,11 +1208,33 @@ namespace smt::noodler {
             }
             expr_ref substr_in(m_util_s.re.mk_in_re(v, substr_re), m);
 
+            string_theory_propagation(xey);
             // 0 <= i <= |s| && 0 <= l <= |s| - i -> |v| in substr_re
             add_axiom({~i_ge_0, ~ls_le_i, ~l_ge_zero, ~li_ge_ls, mk_eq(le, l, false)});
             add_axiom({~i_ge_0, ~ls_le_i, ~l_ge_zero, ~li_ge_ls, mk_literal(substr_in)});
             // 0 <= i <= |s| && |s| < l + i  -> s = x.v
```

Comment:
Is this comment still correct?

Reply:
Yes the comments are bit messy, I have fixed them.